src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -284,13 +287,28 @@ private void ChangeProjectProperty<T>(
 
                     if (!isFullyLoaded)
                     {
-                        TryReportCompilationThrownAway(_projectSystemProjectFactory.Workspace.CurrentSolution, Id);
+                        var reportCompilationThrownAway = GetReportCompilationThrownAway(newValue, oldValue);
+
+                        if (reportCompilationThrownAway)
+                            TryReportCompilationThrownAway(_projectSystemProjectFactory.Workspace.CurrentSolution, Id);
                     }
                 }
             }
+        }
 
-            _projectPropertyModificationsInBatch.Add(
-                (solutionChanges, projectUpdateState) => updateSolution(solutionChanges, projectUpdateState, oldValue));
+        bool GetReportCompilationThrownAway(T newValue, T? oldValue)
+        {
+            // Preprocessor directive only ParseOptions changes have special handling in DocumentState.UpdateParseOptionsAndSourceCodeKind.
+            // We don't want to report telemetry for those changes or for the initial evaluation result.
+            if (newValue is not ParseOptions newParseOption)
```

Comment:
consider doc'ing the two flows that get here.  one that is a ParsOptions one that is not.  this was not clear to me :D